modules/datastore/src/Storage/DatabaseTable.php

```diff
@@ -174,12 +174,22 @@ public function innodbStrictMode(bool $on) {
   }
 
   /**
-   * Get table schema.
+   * Get the table schema in Drupal Schema API format.
    *
-   * @todo Note that this will break on PostgresSQL
+   * NOTE: This will likely fail on any db driver other than mysql.
+   *
+   * @param string $tableName
+   *   The table name.
+   * @param array $fieldsInfo
+   *   Array of fields info from DESCRIBE query.
+   *
+   * @return array
+   *   Full Drupal Schema API array.
    */
-  protected function buildTableSchema($tableName, $fieldsInfo) {
+  protected function buildTableSchema(string $tableName, array $fieldsInfo) {
```

Comment:
just adding a return type
```suggestion
  protected function buildTableSchema(string $tableName, array $fieldsInfo): array {
```